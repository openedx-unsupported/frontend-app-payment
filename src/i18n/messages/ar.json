{
  "payment.error.fetch.basket": "حدثت مشكلة غير متوقعة، إن استمرت المشكلة فيرجى {supportLink}.",
  "payment.error.fetch.basket.support.fragment": "الاتصال بالدعم",
  "payment.messages.enrollment-code-product-info.header": "أتشتري لنفسك فقط؟",
  "payment.messages.enrollment-code-product-info.body": "إن كنت ستشتري رمزًا واحدًا لشخص آخر، فيرجى المواصلة ﻹتمام الطلبية. أما إن كنت أنت المتعلم، ف{link}.",
  "payment.messages.enrollment-code-product-info.link": "انفر هنا للتسجيل مباشرة",
  "payment.messages.enrollment.code.product.info.quantity.updated.header": "لقد حدّثنا كميتك.",
  "payment.messages.enrollment.code.product.info.quantity.updated.body": "في سلتك {quantity, plural,\n   one {رمز تسجيل واحد}\n   two {رمزان للتسجيل}\n   few {# رموز تسجيل}\n   many {# رمز تسجيل}\n   other {# رمز تسجيل}\n} تقدّر تكلفتها بـ {price}, ستصلك عبر البريد الإلكتروني.",
  "payment.messages.transaction.declined.body": "لم نستطع معالجة دفعتك، رجاءً تحقق من معلوماتك الخاصة بالدفع أو تقرب من مصرفك أو مؤسستك المالية من أجل المساعدة.",
  "payment.messages.transaction.error.basket_changed": "Your cart has changed since navigating to this page. Please reload the page and verify the product you are purchasing.",
  "payment.messages.key.timeout.body": "من أجل الأمان، ستحتاج مجددًا لإدخال معلومات بطاقة ائتمانك لإتمام الشراء.",
  "payment.messages.key.timeout.2mins.header": "يرجى إتمام الشراء في دقيقتين.",
  "payment.messages.key.timeout.1min.header": "يرجى إتمام الشراء في دقيقة واحدة.",
  "payment.bulk.summary.table.label.price": "السعر",
  "payment.bulk.summary.table.label.quantity": "الكمّية",
  "payment.bulk.summary.table.label.subtotal": "المجموع الفرعي",
  "payment.screen.reader.cart.details.loaded": "يتم الآن تحميل تفاصيل سلّة التسوّق.",
  "payment.section.cart.label": "تفاصيل سلّة التسوق",
  "payment.productlineitem.purchase.cart.heading": "في سلتك",
  "payment.productlineitem.purchase.cart.subheading": "تتضمن مشترياتك ما يلي:",
  "payment.screen.reader.cart.details.loading": "التحميل جارٍ، يرجى الانتظار...",
  "payment.coupon.label": "إضافة رمز قسيمة (اختياري)",
  "payment.coupon.submit": "تطبيق",
  "payment.coupon.benefit.absolute": "تم تطبيق القسيمة {code} لخصم {amount}",
  "payment.coupon.benefit.percentage": "تم تطبيق القسيمة {code} لخصم {amount}‏%",
  "payment.coupon.benefit.default": "تم تطبيق القسيمة {code}",
  "payment.coupon.remove": "إزالة",
  "payment.currency.disclaimer": "* يجتوي هذا المجموع تحويلًا تقريبا، ستدفع {actualAmount} {actualCurrencyCode}.",
  "payment.summary.discount.offer": "خصم {benefit} مقدم من {provider}.",
  "payment.summary.discount.dynamic_offer": "تم تطبيق خصم {benefit} لقاء ترقيتك الأولى.",
  "payment.summary.table.label.discount.total": "الخصوم المطبقة",
  "payment.order.details.course.entitlement": "عند إتمام طلبيتك ستكون قادرًا على تحديد تواريخ المساق من لوحة معلوماتك.",
  "payment.order.details.course.seat.verified": "The above total includes any applicable taxes. After you complete your order you will be automatically enrolled in the verified track of the course.",
  "payment.order.details.course.seat.credit": "عند إتمام طلبيتك ستحصل على اعتماد لمساقك.",
  "payment.order.details.course.seat": "عند استكمال طلبيتك سيتم إلحاقك تلقائيًا في للمساق.",
  "payment.order.details.enrollment.code.terms": "عند إتمام الطلب، فإنك توافق أنت ومؤسستك على الشروط التالية:",
  "payment.order.details.enrollment.code.first.term": "كل رمز يشمل مساقًا واحدًا، ويمكن استخدامه مرة واحدة فقط.",
  "payment.order.details.enrollment.code.second.term": "أنت مسؤل عن توزيع رموز القسائم لمتعلّميك في مؤسستك",
  "payment.order.details.enrollment.code.third.term": "ستنتهي صلاحية كل رمز خلال سنة من تاريخ الشراء أو عند إغلاق المساق -على الأقرب.",
  "payment.order.details.enrollment.code.fourth.term": "إذا لم يتم ضبط إعداد المساق بنظام التعلم الذاتي، فيجب أن تتأكد من توفر مشغل المساق قبل انتهاء صلاحيته.",
  "payment.order.details.enrollment.code.fifth.term": "لا يحق لك إعادة بيع رموز القسائم لأطراف خارجية.",
  "payment.order.details.enrollment.code.sixth.term": "جميع عمليات edX هي عمليات تجارية نهائية وغير مخولة لاسترداد الرسوم.",
  "payment.order.details.enrollment.code.receive.email": "ستتلقى رسالة بريد إلكتروني على {userEmail} برموز التحاقك.",
  "payment.order.details.heading": "ملخّص",
  "payment.productlineitem.professional.certificate": "شهادة مهنية",
  "payment.productlineitem.verified.certificate": "شهادة معتمدة",
  "payment.summary.table.label.price": "السعر",
  "payment.summary.table.label.total.to.pay": "إجمالي",
  "payment.update.quantity.label": "الكمّية",
  "payment.update.quantity.field.help.text": "الحد الأقصى: {number}",
  "payment.update.quantity.submit.button": "تحديث",
  "payment.select.payment.method.heading": "اختر طريقة الدفع",
  "payment.section.payment.details.label": "تفاصيل الدفع",
  "payment.page.method.type.credit": "بطاقة الائتمان",
  "payment.page.method.type.paypal": "باي بال",
  "payment.form.submit.button.text": "إيداع الطلبية",
  "payment.card.details.billing.information.heading": "معلومات الفوترة (مطلوبة)",
  "payment.card.details.expiration.month.label": "شهر انتهاء الصلاحية (مطلوب)",
  "payment.card.details.expiration.year.label": "سنة انتهاء الصلاحية (مطلوب)",
  "payment.page.card.details.month": "الشهر",
  "payment.page.card.details.year": "السنة",
  "payment.card.holder.information.postal.code.label.required": "الرمز البريدي (مطلوب)",
  "payment.card.holder.information.postal.code.label": "الرمز البريدي",
  "payment.card.holder.information.heading": "معلومات حامل البطاقة",
  "payment.card.holder.information.first.name.label": "الاسم الأول (مطلوب)",
  "payment.card.holder.information.last.name.label": "الاسم الأخير (مطلوب)",
  "payment.card.holder.information.organization.label": "المؤسسة (مطلوب)",
  "payment.card.holder.information.address.label": "العنوان (مطلوب)",
  "payment.card.holder.information.unit.label": "رقم الجناح/الشقة",
  "payment.card.holder.information.city.label": "المدينة (مطلوب)",
  "payment.card.holder.information.country.label": "الدولة (مطلوب)",
  "payment.card.holder.information.purchased.for.organization": "أشتري نيابة عن مديري أو أي شخص في مؤسستي المهنية.",
  "payment.card.holder.information.country.options.empty": "اختر البلد",
  "payment.card.details.number.label": "رقم البطاقة (مطلوب)",
  "payment.card.details.security.code.label": "رمز الأمان (مطلوب)",
  "payment.card.details.security.code.help.text": "آخر ثلاثة أرقام في منطقة التوقيع الموجودة على ظهر البطاقة الخاصة بك. بالنسبة لبطاقة أمريكان إكسبرس، يوجد أربعة أرقام على وجه البطاقة.",
  "payment.form.errors.invalid.card.number": "رقم بطاقة غير صحيح",
  "payment.form.errors.unsupported.card": "نوع بطاقة غير مدعوم",
  "payment.form.errors.invalid.security.code": "رمز أمان غير صحيح",
  "payment.form.errors.card.expired": "البطاقة منتهية الصلاحية",
  "payment.form.errors.required.field": "هذا الحقل مطلوب",
  "payment.form.errors.ascii.name": "نأسف للإزعاج، لكن حقل الاسم يقبل حروف ASCII فقط. نحن نعمل على معالجة هذا الأمر ونقدر سعة صدرك.",
  "payment.card.holder.information.state.label": "الولاية/المقاطعة (مطلوبة)",
  "payment.card.holder.information.state.required.label": "الولاية/المقاطعة",
  "payment.card.holder.information.state.options.empty": "اختر الولاية/المنطقة",
  "payment.empty.basket.heading": "سلتك فارغة.",
  "payment.empty.basket.message": "إذا حاولت إجراء عملية شراء، فلن يتم فرض رسوم عليك. عُد إلى {actionLinkOne} للمحاولة مرة أخرى، أو {actionLinkTWO}.",
  "payment.empty.basket.dashboardURL": "لوحة معلومات",
  "payment.empty.basket.supportURL": "اتصل بفريق edX لدعم التجارة الإلكترونية",
  "payment.apple.pay.pay.with.apple.pay": "الدفع بواسطة  Apple Pay",
  "payment.type.paypal": "باي بال",
  "payment.heading.page": "الدفع",
  "payment.loading.payment": "جاري تحميل السلّة...",
  "payment.loading.error": "خطأ: {error}",
  "payment.apple.pay.merchant.validation.failure": "Apple Pay غير متوفر في هذا الوقت. الرجاء تجربة طريقة دفع أخرى.",
  "payment.apple.pay.authorization.failure": "حدث خطأ أثناء معالجة عملية الدفع. لم يتم تحصيل رسوم منك. الرجاء إعادة المحاولة، أو اختيار طريقة أخرى للدفع.",
  "subscription.alerts.error.embargo": "We're sorry, this program is not available in your region.",
  "subscription.alerts.error.program_unavailable": "Something went wrong, please reload the page. If the issue persists please {supportLink}.",
  "subscription.alerts.error.ineligible_program": "We're sorry, this program is no longer offering a subscription option. Please search our catalog for current availability",
  "subscription.checkout.payment.label": "Subscription payment details.",
  "subscription.checkout.billing.notification": "You will be charged {price} {currency} {trialEnd} then monthly until you cancel your subscription.",
  "subscription.checkout.billing.trial.date": "on {date},",
  "subscription.checkout.billing.resubscribe.date": "today,",
  "subscription.checkout.form.submit.button.text.default": "Subscribe",
  "subscription.checkout.form.submit.button.text.resubscribe": "Subscription started",
  "subscription.confirmation.modal.navigation.title": "Go to dashboard",
  "subscription.confirmation.modal.trialing.heading": "Congratulations! Your 7-day free trial of {programTitle} has started.",
  "subscription.confirmation.modal.trialing.body": "When your free trial ends, your subscription will begin, and we'll charge your payment method on file {price} {currency} per month. To avoid being charged, you must cancel before your trial expires. This subscription will automatically renew every month unless you cancel from the {ordersAndSubscriptionLink} page.",
  "subscription.confirmation.modal.resubscribe.heading": "Congratulations! Your subscription to {programTitle} has started.",
  "subscription.confirmation.modal.resubscribe.body": "We charged your payment method {price} {currency}. This subscription will be automatically renewed and charged monthly unless you cancel from the {ordersAndSubscriptionLink} page.",
  "subscription.confirmation.modal.body.orders.link": "Orders and Subscriptions",
  "subscription.purchase.details.heading": "In your cart",
  "subscription.purchase.details.label": "Subscription",
  "subscription.purchase.details.product.list.heading": "Included with your subscription:",
  "subscription.details.order.legal": "By signing up for a subscription or starting a free trial, you authorize us to charge your card on file {price}/month {currency} for {programTitle} and any applicable taxes. Your {programTitle} subscriptions will automatically renew until you cancel. Monthly subscriptions can be canceled at any time, review the cancellation process on the {supportLink}. Canceling a subscription stops the monthly recurring subscription charge, but doesn't refund the transaction for the current billing period. You can continue to access the subscription until the end of the then current billing period.",
  "subscription.details.order.legal.link": "Learner Help Center",
  "subscription.order.details": "After you complete your order you will have verified access to each course in {programTitle}.",
  "subscription.order.details.heading": "Order details",
  "subscription.order.details.taxes": "The above total includes any applicable taxes.",
  "subscription.order.details.cancel": "Cancel anytime.",
  "subscription.details.program.type.microMasters": "MicroMasters",
  "subscription.details.program.type.microBachelors": "MicroBachelors",
  "subscription.details.program.type.xSeries": "XSeries",
  "subscription.details.program.type.professionalProgram": "Professional Program",
  "subscription.details.program.type.professionalCertificate": "Professional Certificate",
  "subscription.details.program.type.masters": "Masters",
  "subscription.screen.reader.details.loading": "Loading, please wait...",
  "subscription.screen.reader.details.loaded": "Shopping details are loaded.",
  "subscription.summary.table.label.price": "Price",
  "subscription.summary.table.label.trialing.price": "/month USD after 7-day free trial",
  "subscription.summary.table.label.resubscribe.price": "/month USD",
  "subscription.summary.table.label.total.to.pay": "Today's total",
  "subscription.heading.page": "Subscription Payment",
  "subscription.loading.details": "Loading subscription..."
}