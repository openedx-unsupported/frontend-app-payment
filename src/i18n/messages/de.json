{
  "payment.error.fetch.basket": "",
  "payment.error.fetch.basket.support.fragment": "Technischer Support",
  "payment.messages.enrollment-code-product-info.header": "",
  "payment.messages.enrollment-code-product-info.body": "",
  "payment.messages.enrollment-code-product-info.link": "",
  "payment.messages.enrollment.code.product.info.quantity.updated.header": "",
  "payment.messages.enrollment.code.product.info.quantity.updated.body": "",
  "payment.messages.transaction.declined.body": "",
  "payment.messages.transaction.error.basket_changed": "",
  "payment.messages.key.timeout.body": "",
  "payment.messages.key.timeout.2mins.header": "",
  "payment.messages.key.timeout.1min.header": "",
  "payment.empty.basket.heading": "",
  "payment.empty.basket.message": "",
  "payment.empty.basket.dashboardURL": "",
  "payment.empty.basket.supportURL": "",
  "payment.heading.page": "Zahlung",
  "payment.loading.payment": "",
  "payment.loading.error": "",
  "payment.apple.pay.merchant.validation.failure": "",
  "payment.apple.pay.authorization.failure": "",
  "subscription.heading.page": "",
  "subscription.loading.details": "",
  "payment.bulk.summary.table.label.price": "Preis",
  "payment.bulk.summary.table.label.quantity": "",
  "payment.bulk.summary.table.label.subtotal": "",
  "payment.screen.reader.cart.details.loaded": "",
  "payment.section.cart.label": "",
  "payment.productlineitem.purchase.cart.heading": "",
  "payment.productlineitem.purchase.cart.subheading": "",
  "payment.screen.reader.cart.details.loading": "",
  "payment.coupon.label": "",
  "payment.coupon.submit": "Apply",
  "payment.coupon.benefit.absolute": "",
  "payment.coupon.benefit.percentage": "",
  "payment.coupon.benefit.default": "",
  "payment.coupon.remove": "Entfernen",
  "payment.currency.disclaimer": "",
  "payment.summary.discount.offer": "",
  "payment.summary.discount.dynamic_offer": "",
  "payment.summary.table.label.discount.total": "",
  "payment.order.details.heading": "Bestellungsdetails",
  "payment.productlineitem.professional.certificate": "",
  "payment.productlineitem.verified.certificate": "",
  "payment.summary.table.label.price": "Preis",
  "payment.summary.table.label.total.to.pay": "",
  "payment.update.quantity.label": "",
  "payment.update.quantity.field.help.text": "",
  "payment.update.quantity.submit.button": "Aktualisieren",
  "payment.select.payment.method.heading": "",
  "payment.section.payment.details.label": "",
  "payment.page.method.type.credit": "",
  "payment.page.method.type.paypal": "",
  "payment.form.submit.button.text": "",
  "subscription.alerts.error.embargo": "",
  "subscription.alerts.error.program_unavailable": "",
  "subscription.alerts.error.ineligible_program": "",
  "subscription.alerts.error.requires_payment_method": "",
  "subscription.checkout.payment.label": "",
  "subscription.confirmation.modal.navigation.title": "Zum Dashboard",
  "subscription.confirmation.modal.trialing.heading": "",
  "subscription.confirmation.modal.trialing.body": "",
  "subscription.confirmation.modal.resubscribe.heading": "",
  "subscription.confirmation.modal.resubscribe.body": "",
  "subscription.confirmation.modal.body.orders.link": "",
  "subscription.screen.reader.details.loaded": "",
  "payment.order.details.enrollment.code.terms": "",
  "payment.order.details.enrollment.code.first.term": "",
  "payment.order.details.enrollment.code.second.term": "",
  "payment.order.details.enrollment.code.third.term": "",
  "payment.order.details.enrollment.code.fourth.term": "",
  "payment.order.details.enrollment.code.fifth.term": "",
  "payment.order.details.enrollment.code.sixth.term": "",
  "payment.order.details.enrollment.code.receive.email": "",
  "payment.order.details.course.entitlement": "",
  "payment.order.details.course.seat.verified": "",
  "payment.order.details.course.seat.credit": "",
  "payment.order.details.course.seat": "",
  "payment.card.details.billing.information.heading": "",
  "payment.card.details.expiration.month.label": "",
  "payment.card.details.expiration.year.label": "",
  "payment.page.card.details.month": "",
  "payment.page.card.details.year": "",
  "payment.card.holder.information.postal.code.label.required": "",
  "payment.card.holder.information.postal.code.label": "",
  "payment.card.holder.information.heading": "",
  "payment.card.holder.information.first.name.label": "",
  "payment.card.holder.information.last.name.label": "",
  "payment.card.holder.information.organization.label": "",
  "payment.card.holder.information.address.label": "",
  "payment.card.holder.information.unit.label": "",
  "payment.card.holder.information.city.label": "",
  "payment.card.holder.information.country.label": "",
  "payment.card.holder.information.purchased.for.organization": "",
  "payment.card.holder.information.country.options.empty": "",
  "payment.form.errors.invalid.card.number": "",
  "payment.form.errors.unsupported.card": "",
  "payment.form.errors.invalid.security.code": "",
  "payment.form.errors.card.expired": "",
  "payment.form.errors.required.field": "",
  "payment.form.errors.ascii.name": "",
  "payment.card.holder.information.state.label": "",
  "payment.card.holder.information.state.required.label": "",
  "payment.card.holder.information.state.options.empty": "",
  "payment.apple.pay.pay.with.apple.pay": "",
  "payment.type.paypal": "",
  "subscription.checkout.billing.notification": "",
  "subscription.checkout.billing.trial.date": "",
  "subscription.checkout.billing.resubscribe.date": "",
  "subscription.checkout.form.submit.button.text.default": "",
  "subscription.checkout.form.submit.button.text.resubscribe": "",
  "subscription.purchase.details.heading": "",
  "subscription.purchase.details.label": "",
  "subscription.purchase.details.product.list.heading": "",
  "subscription.details.order.legal": "",
  "subscription.details.order.legal.link": "",
  "subscription.order.details": "",
  "subscription.order.details.heading": "",
  "subscription.order.details.taxes": "",
  "subscription.order.details.cancel": "",
  "subscription.details.program.type.microMasters": "",
  "subscription.details.program.type.microBachelors": "",
  "subscription.details.program.type.xSeries": "",
  "subscription.details.program.type.professionalProgram": "",
  "subscription.details.program.type.professionalCertificate": "",
  "subscription.details.program.type.masters": "",
  "subscription.screen.reader.details.loading": "",
  "subscription.summary.table.label.price": "Preis",
  "subscription.summary.table.label.trialing.price": "",
  "subscription.summary.table.label.resubscribe.price": "",
  "subscription.summary.table.label.total.to.pay": "",
  "payment.card.details.number.label": "",
  "payment.card.details.security.code.label": "",
  "payment.card.details.security.code.help.text": ""
}