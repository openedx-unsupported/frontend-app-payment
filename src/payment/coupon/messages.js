import { defineMessages } from '@edx/frontend-i18n';

const messages = defineMessages({
  'payment.coupon.label': {
    id: 'payment.coupon.label',
    defaultMessage: 'Add coupon code (optional)',
    description: 'Label for the add coupon form',
  },
  'payment.coupon.submit': {
    id: 'payment.coupon.submit',
    defaultMessage: 'Apply',
    description: 'Submit button for the add coupon form',
  },
  'payment.coupon.remove': {
    id: 'payment.coupon.remove',
    defaultMessage: 'Remove',
    description: 'Submit button to remove a coupon',
  },
  'payment.coupon.benefit_value': {
    id: 'payment.coupon.benefit_value',
    defaultMessage: 'Coupon {code} applied for {value} off',
    description: 'The description of a coupon for a percentage off the total price. Percentage symbol is included in {value}',
  },
  'payment.coupon.benefit.default': {
    id: 'payment.coupon.benefit.default',
    defaultMessage: 'Coupon {code} applied',
    description: 'The description of a coupon without any specific monetary value off.',
  },
});

export default messages;
