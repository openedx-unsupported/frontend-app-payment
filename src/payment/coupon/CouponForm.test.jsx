import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { IntlProvider, injectIntl, configure as configureI18n } from '@edx/frontend-i18n';

import { configuration } from '../../environment';
import messages from '../../i18n';

import ConnectedCouponForm, { CouponForm } from './CouponForm';

jest.mock('@edx/frontend-logging', () => ({
  logError: jest.fn(),
}));

configureI18n(configuration, messages);

const mockStore = configureMockStore();
const storeMocks = {
  defaultState: require('./__mocks__/defaultCouponState.mockStore.js'), // eslint-disable-line global-require
  percentageCouponAdded: require('./__mocks__/percentageCouponAdded.mockStore.js'), // eslint-disable-line global-require
  absoluteCouponAdded: require('./__mocks__/absoluteCouponAdded.mockStore.js'), // eslint-disable-line global-require
  defaultCouponAdded: require('./__mocks__/defaultCouponAdded.mockStore.js'), // eslint-disable-line global-require
  couponEntered: require('./__mocks__/couponEntered.mockStore.js'), // eslint-disable-line global-require
};

describe('CouponForm', () => {
  it('should render the add coupon form', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider store={mockStore(storeMocks.defaultState)}>
          <ConnectedCouponForm
            addCoupon={jest.fn()}
            removeCoupon={jest.fn()}
            updateCouponDraft={jest.fn()}
          />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a percentage coupon', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider store={mockStore(storeMocks.percentageCouponAdded)}>
          <ConnectedCouponForm
            addCoupon={jest.fn()}
            removeCoupon={jest.fn()}
            updateCouponDraft={jest.fn()}
          />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an absolute coupon', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider store={mockStore(storeMocks.absoluteCouponAdded)}>
          <ConnectedCouponForm
            addCoupon={jest.fn()}
            removeCoupon={jest.fn()}
            updateCouponDraft={jest.fn()}
          />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('without logging errors', () => {
    /**
     * Explanation: Why do this?  This test is in anticipation of there being coupon types we
     * didn't expect.  We want the UI to at least display a message in these cases, even if we
     * weren't expecting them, because the alternative is to show no text at all with a "Remove"
     * button referencing nothing.  Because this is the basket page, we want to be
     * over-cautious and let the UI support it in a minimal way.
     *
     * We want the Benefit.jsx component to continue to warn us about unexpected types in
     * development, but we want to have good fallback behavior in prod, where the console.errors
     * won't show up.  Therefore, we're hiding the errors here so that we don't have developers
     * seeing false positives on test failures (when they aren't actually failures!)
     */
    let originalError;

    beforeEach(() => {
      originalError = console.error; // eslint-disable-line no-console
      console.error = jest.fn(); // eslint-disable-line no-console
    });

    afterEach(() => {
      console.error = originalError; // eslint-disable-line no-console
    });

    it('should render an unexpected coupon', () => {
      const component = (
        <IntlProvider locale="en">
          <Provider store={mockStore(storeMocks.defaultCouponAdded)}>
            <ConnectedCouponForm
              addCoupon={jest.fn()}
              removeCoupon={jest.fn()}
              updateCouponDraft={jest.fn()}
            />
          </Provider>
        </IntlProvider>
      );
      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('should call addCoupon on form submit', () => {
    const addCoupon = jest.fn();
    const IntlCouponForm = injectIntl(CouponForm);
    const wrapper = mount((
      <IntlProvider locale="en">
        <IntlCouponForm
          addCoupon={addCoupon}
          removeCoupon={jest.fn()}
          updateCouponDraft={jest.fn()}
          code="DEMO25"
        />
      </IntlProvider>
    ));

    wrapper.find('form').simulate('submit');
    expect(addCoupon).toHaveBeenCalledWith('DEMO25');
  });

  it('should call removeCoupon on form submit', () => {
    const removeCoupon = jest.fn();
    const IntlCouponForm = injectIntl(CouponForm);
    const wrapper = mount((
      <IntlProvider locale="en">
        <IntlCouponForm
          addCoupon={jest.fn()}
          removeCoupon={removeCoupon}
          updateCouponDraft={jest.fn()}
          code="DEMO25"
          voucherId={12345}
        />
      </IntlProvider>
    ));

    wrapper.find('form').simulate('submit');
    expect(removeCoupon).toHaveBeenCalledWith(12345);
  });

  it('should call updateCouponDraft on code change', () => {
    const updateCouponDraft = jest.fn();
    const IntlCouponForm = injectIntl(CouponForm);
    const wrapper = mount((
      <IntlProvider locale="en">
        <IntlCouponForm
          addCoupon={jest.fn()}
          removeCoupon={jest.fn()}
          updateCouponDraft={updateCouponDraft}
          code="DEMO25"
        />
      </IntlProvider>
    ));

    wrapper.find('form').find('input').simulate('change', { target: { value: 'DEMO' } });
    expect(updateCouponDraft).toHaveBeenCalledWith('DEMO');
  });

  describe('error handling', () => {
    const IntlCouponForm = injectIntl(CouponForm);

    function createComponent(errorCode) {
      return (
        <IntlProvider locale="en">
          <IntlCouponForm
            addCoupon={jest.fn()}
            removeCoupon={jest.fn()}
            updateCouponDraft={jest.fn()}
            code="DEMO25"
            errorCode={errorCode}
          />
        </IntlProvider>
      );
    }

    it('should match error state snapshot', () => {
      const tree = renderer.create(createComponent('unknown_error')).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should display empty_basket errors correctly', () => {
      const wrapper = mount(createComponent('empty_basket'));
      expect(wrapper.find('strong.invalid-feedback').text()).toMatchSnapshot();
    });

    it('should display already_applied_voucher errors correctly', () => {
      const wrapper = mount(createComponent('already_applied_voucher'));
      expect(wrapper.find('strong.invalid-feedback').text()).toMatchSnapshot();
    });

    it('should display code_does_not_exist errors correctly', () => {
      const wrapper = mount(createComponent('code_does_not_exist'));
      expect(wrapper.find('strong.invalid-feedback').text()).toMatchSnapshot();
    });

    it('should display code_expired errors correctly', () => {
      const wrapper = mount(createComponent('code_expired'));
      expect(wrapper.find('strong.invalid-feedback').text()).toMatchSnapshot();
    });

    it('should display code_not_active errors correctly', () => {
      const wrapper = mount(createComponent('code_not_active'));
      expect(wrapper.find('strong.invalid-feedback').text()).toMatchSnapshot();
    });

    it('should display code_not_available errors correctly', () => {
      const wrapper = mount(createComponent('code_not_available'));
      expect(wrapper.find('strong.invalid-feedback').text()).toMatchSnapshot();
    });

    it('should display code_not_valid errors correctly', () => {
      const wrapper = mount(createComponent('code_not_valid'));
      expect(wrapper.find('strong.invalid-feedback').text()).toMatchSnapshot();
    });

    it('should display unknown errors correctly', () => {
      const wrapper = mount(createComponent('some_other_error'));
      expect(wrapper.find('strong.invalid-feedback').text()).toMatchSnapshot();
    });
  });
});
