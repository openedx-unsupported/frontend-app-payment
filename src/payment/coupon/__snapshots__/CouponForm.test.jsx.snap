// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponForm should render a percentage coupon 1`] = `
<form
  className="d-flex align-items-center mb-3"
  onSubmit={[Function]}
>
  <span>
    Coupon DEMO25 applied for 25% off
  </span>
  <button
    className="btn btn-link display-inline p-0 pl-3 border-0"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    type="submit"
  >
    Remove
  </button>
</form>
`;

exports[`CouponForm should render an absolute coupon 1`] = `
<form
  className="d-flex align-items-center mb-3"
  onSubmit={[Function]}
>
  <span>
    Coupon DEMO30 applied for $30 off
  </span>
  <button
    className="btn btn-link display-inline p-0 pl-3 border-0"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    type="submit"
  >
    Remove
  </button>
</form>
`;

exports[`CouponForm should render the add coupon form 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="h6 d-block"
      htmlFor="couponField"
    >
      Add coupon code (optional)
    </label>
    <input
      aria-describedby=""
      className="form-control mb-2"
      id="couponField"
      name="couponField"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="submit"
    >
      Apply
    </button>
  </div>
</form>
`;

exports[`CouponForm without logging errors should render an unexpected coupon 1`] = `
<form
  className="d-flex align-items-center mb-3"
  onSubmit={[Function]}
>
  <span>
    Coupon DEMO25 applied
  </span>
  <button
    className="btn btn-link display-inline p-0 pl-3 border-0"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    type="submit"
  >
    Remove
  </button>
</form>
`;
